pkg/monitor/monitor/template.go

```diff
@@ -162,15 +98,184 @@ type MonitorConfigModel struct {
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func addTlsConfig(pc *PrometheusConfig, model *MonitorConfigModel) {
+	for id, config := range pc.ScrapeConfigs {
+		if config.JobName == "tidb-cluster" {
+			config.TlsConfig.CaFile = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+			config.TlsConfig.CertFile = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+			config.TlsConfig.KeyFile = "/var/lib/pd-client-tls/key"
+			config.Schema = "https"
+			// This is a workaround of https://github.com/tikv/tikv/issues/5340 and should
+			// be removed after TiKV fix this issue
+			config.RelabelConfigs = append(config.RelabelConfigs, RelabelConfig{
+				SourceLabels: "[__meta_kubernetes_pod_name]",
+				Action:       "drop",
+				Regex:        ".*\\-tikv\\-\\d*$",
+			})
+		}
+		pc.ScrapeConfigs[id] = config
+	}
+	// This is a workaround of https://github.com/tikv/tikv/issues/5340 and should
+	// be removed after TiKV fix this issue
+	pc.ScrapeConfigs = append(pc.ScrapeConfigs, ScrapeConfigSpec{
+		JobName:        "tidb-cluster-tikv",
+		ScrapeInterval: "15s",
+		HonorLabels:    true,
+		KubernetesSDConfigs: []KubernetesSDConfig{
+			{
+				Role: "pod",
+				Namespaces: &NamespacesConfig{
+					Names: model.ReleaseNamespaces,
+				},
+			},
+		},
+		TlsConfig: TlsConfig{
+			InsecureSkipVerify: true,
+		},
+		Schema: "http",
+		RelabelConfigs: []RelabelConfig{
+			{
+				SourceLabels: "[__meta_kubernetes_pod_label_app_kubernetes_io_instance]",
+				Action:       "keep",
+				Regex:        model.ReleaseTargetRegex,
+			},
+			{
+				SourceLabels: "[__meta_kubernetes_pod_annotation_prometheus_io_scrape]",
+				Action:       "keep",
+				Regex:        "true",
+			},
+			{
+				SourceLabels: "[__meta_kubernetes_pod_annotation_prometheus_io_path]",
+				Action:       "replace",
+				TargetLabel:  "__metrics_path__",
+				Regex:        "(.+)",
+			},
+			{
+				SourceLabels: "[__address__, __meta_kubernetes_pod_annotation_prometheus_io_port]",
+				Action:       "replace",
+				Regex:        "([^:]+)(?::\\d+)?;(\\d+)",
+				Replacement:  "$1:$2",
+				TargetLabel:  "__address__",
+			},
+			{
+				SourceLabels: "[__meta_kubernetes_namespace]",
+				Action:       "replace",
+				TargetLabel:  "kubernetes_namespace",
+			},
+			{
+				SourceLabels: "[__meta_kubernetes_pod_node_name]",
+				Action:       "replace",
+				TargetLabel:  "kubernetes_node",
+			},
+			{
+				SourceLabels: "[__meta_kubernetes_pod_ip]",
+				Action:       "replace",
+				TargetLabel:  "kubernetes_pod_ip",
+			},
+		},
+	})
+}
+
+func RenderPrometheusConfig(model *MonitorConfigModel) (string, error) {
+	pc := newPrometheusConfig(model)
+	if model.EnableTLSCluster {
+		addTlsConfig(pc, model)
+	}
+	if len(model.AlertmanagerURL) > 0 {
+		addAlertManagerUrl(pc, model)
+	}
+	bs, err := yaml.Marshal(pc)
 	if err != nil {
 		return "", err
 	}
-	return buff.String(), nil
+	// remove character "'"
+	return strings.ReplaceAll(string(bs), "'", ""), nil
```

Comment:
why is this necessary?

A suggestion for good comment is to comment the intent of the code, not the things it does

Reply:
removed now.